salt/cloud/__init__.py

```diff
@@ -2083,128 +2085,145 @@ def run_map(self, dmap):
 
         # Now sort the create list based on dependencies
         create_list = sorted(six.iteritems(dmap['create']), key=lambda x: x[1]['level'])
+        full_map = dmap['create'].copy()
+        full_map.update(dmap['existing'])
+        possible_master_list = sorted(six.iteritems(full_map), key=lambda x: x[1]['level'])
         output = {}
         if self.opts['parallel']:
             parallel_data = []
         master_name = None
         master_minion_name = None
         master_host = None
         master_finger = None
-        try:
-            master_name, master_profile = next((
-                (name, profile) for name, profile in create_list
-                if profile.get('make_master', False) is True
-            ))
-            master_minion_name = master_name
-            log.debug('Creating new master \'{0}\''.format(master_name))
-            if salt.config.get_cloud_config_value(
-                'deploy',
-                master_profile,
-                self.opts
-            ) is False:
-                raise SaltCloudSystemExit(
-                    'Cannot proceed with \'make_master\' when salt deployment '
-                    'is disabled(ex: --no-deploy).'
-                )
-
-            # Generate the master keys
-            log.debug(
-                'Generating master keys for \'{0[name]}\''.format(master_profile)
-            )
-            priv, pub = salt.utils.cloud.gen_keys(
-                salt.config.get_cloud_config_value(
-                    'keysize',
+        for name, profile in possible_master_list:
+            if profile.get('make_master', False) is True:
+                master_name = name
+                master_profile = profile
+
+        if master_name:
+            # If the master already exists, get the host
+            if master_name not in dmap['create']:
+                cloud_client = salt.cloud.CloudClient()
+                master_host = cloud_client.query()
```

Comment:
Why are you initializing a cloud client? you are already in one, just run self.query().

Reply:
There's no .query() on self.

Reply:
Oh, right, this is in the map, ok, what you should do is add

```
self.client = CloudClient(self.opts)
```

right here https://github.com/saltstack/salt/blob/develop/salt/cloud/__init__.py#L519

and then add self.client.query()